src/core_plugins/kibana/public/management/react/store/actions/indices.js

```diff
@@ -0,0 +1,63 @@
+import { globals } from '../../globals';
+import { endsWith, startsWith, uniq, sortBy } from 'lodash';
+import { createAction } from 'redux-actions';
+import { createThunk } from 'redux-thunks';
+
+import { fetchTimeFields } from './time_fields';
+
+export const fetchedIndices = createAction('FETCHED_INDICES',
+  (indices, pattern, hasExactMatches) => ({ indices, pattern, hasExactMatches }));
+
+export const fetchIndices = createThunk('FETCH_INDICES',
+  async ({ dispatch }, pattern, initialFetch = false) => {
```

Comment:
++ to the -1, and it causes a hard to debug error with the optimizer.

Reply:
Honestly, I'm not really sold on `redux-actions` or `redux-thunks`. I can easily pass on both. Super interested in others thoughts here too

Reply:
I've been really pleased with both, since it patches over the difference between action creators and thunks. Also very interested in how others are doing async in redux though, it's a huge pain any way you slice it.